src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Assembly.NativeAot.cs

```diff
@@ -35,6 +35,16 @@ public static Assembly Load(string assemblyString)
             return Load(name);
         }
 
+        // Performance metric to count the number of assemblies
+        // Caching since in NativeAOT, the number will be the same
+        private static uint s_assemblyCount;
+        internal static uint GetAssemblyCount()
+        {
+            if (s_assemblyCount == 0)
+                s_assemblyCount = (uint)AppDomain.CurrentDomain.GetAssemblies().Length;
```

Comment:
Since we're already in reflection implementation, no need to do the extra work AppDomain does:

```suggestion
                s_assemblyCount = (uint)Internal.Reflection.Core.Execution.ReflectionCoreExecution.ExecutionDomain.ReflectionDomainSetup.AssemblyBinder.GetLoadedAssemblies().Count;
```